plugins/out_s3/s3.c

```diff
@@ -1083,13 +1101,23 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
         final_body = body;
         final_body_size = body_size;
     }
+
+    char final_body_md5[25];
+    memset(final_body_md5, 0, sizeof(final_body_md5));
+    if (ctx->send_content_md5 == FLB_TRUE) {
+        if (s3_body_md5(final_body, final_body_size, final_body_md5, sizeof(final_body_md5)) != 0) {
+            flb_plg_error(ctx->ins, "Failed to MD5 body");
```

Comment:
make the error message very descriptive and formal. Something like "Failed to set content-md5 header" is better.